src/suffixtree.jl

```diff
@@ -0,0 +1,112 @@
+using Base.Test
+import Base.insert!
+
+abstract SuffixTrie
+
+type SuffixTrieNode <: SuffixTrie
+  value
+  children::Array{SuffixTrieNode}
+  index::Int64
+end
+
+SuffixTrieNode(value, children) = SuffixTrieNode(value, children, -1)
+SuffixTrieNode(value) = SuffixTrieNode(value, [], -1)
+
+type SuffixTrieRoot <: SuffixTrie
+  children::Array{SuffixTrieNode}
+end
+
+SuffixTrieRoot() = SuffixTrieRoot([])
```

Comment:
In Julia 0.3, `[]` is an `Array{None,1}` (which is a totally useless type).  In v0.4, it's `Array{Any,1}`, which is more useful, but it might be good to change this if/when you change the types to be parameterized.